### PR TITLE
Create cognitive services

### PR DESCRIPTION
**Pull request by BassOfLion**
Add module for cognitive services

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,74 @@
 # terraform-azurerm-cognitive-services
+
 A terraform module for Azure Cognitive Services such as Face API and Computer Vision
+
+## Usage
+
+Example that shows configuration for deployment of cogtnitive service Face API
```
Comment (yngveh) [inline README.md]: cogtnitive -> cognitive

File: examples/diagnostics/main.tf [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    }
+  }
+
+  tags = {
+    tag1 = "value1"
+  }
+
+  diagnostics = {
+    destination   = "test"
+    eventhub_name = "diagnostics",
```
Comment (yngveh) [inline examples/diagnostics/main.tf]: Remove "," after lines.

File: main.tf [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+resource "azurerm_cognitive_account" "main" {
+  for_each = var.cognitive_services_types
+
+  name                = "${each.value.kind}-cog"
+  location            = azurerm_resource_group.main.location
+  resource_group_name = azurerm_resource_group.main.name
+  kind                = each.value.kind
+  sku_name            = each.value.sku_name
+  tags                = var.tags
+}
```
Comment (yngveh) [inline main.tf:68]: Add network_acls. Should any other attributes be added as well ?

**Review by yngveh — CHANGES_REQUESTED** [review]

File: main.tf (hunk 3 above)
Comment (BassOfLion) [reply main.tf:68]: custom_subdomain_name is required if network_acls is used even though it is specified as optional. Issue registered here https://github.com/hashicorp/terraform-provider-azurerm/issues/10041

**Review by yngveh — APPROVED** [review]

**Pull request merged**